src/lib.rs

```diff
@@ -591,6 +591,11 @@ mod contract {
         );
     }
 
+    #[no_mangle]
+    pub extern "C" fn get_metadata() {
```

Comment:
This function name should be `ft_metadata` according to [the spec](https://nomicon.io/Standards/FungibleToken/Metadata.html#reference-level-explanation).